src/gei/Commands/MigrateRepoCommand.cs

```diff
@@ -243,9 +247,27 @@ private async Task<bool> RepoExists(GithubApi githubApi, string org, string repo
 
         private string GetSourceRepoUrl(MigrateRepoCommandArgs args) =>
             args.GithubSourceOrg.HasValue()
-                ? GetGithubRepoUrl(args.GithubSourceOrg, args.SourceRepo)
+                ? GetGithubRepoUrl(args.GithubSourceOrg, args.SourceRepo, args.GhesApiUrl.HasValue() ? ExtractGhesBaseUrl(args.GhesApiUrl) : null)
                 : GetAdoRepoUrl(args.AdoSourceOrg, args.AdoTeamProject, args.SourceRepo);
 
+        private string ExtractGhesBaseUrl(string ghesApiUrl)
+        {
+            // We expect the GHES url template to be either http(s)://hostname/api/v3 or http(s)://api.hostname.com.
+            // We are either going to be able to extract and return the base url based on the above templates or 
+            // will fallback to ghesApiUrl and return it as the base url. 
+
+            ghesApiUrl = ghesApiUrl.Trim().TrimEnd('/');
+
+            var baseUrl = Regex.Match(ghesApiUrl, @"(?<baseUrl>https?:\/\/.+)\/api\/v3", RegexOptions.IgnoreCase).Groups["baseUrl"].Value;
```

Comment:
does this work for both http and https?

Reply:
yes the `s?` indicates that the last `s` can appear 0 or 1 times.